app/build.gradle.kts

```diff
@@ -92,6 +92,15 @@ android {
 
 play {
   enabled.set(true)
+  val jsonData = System.getenv("google_json")
+  val jsonFile: File = file("../google.json")
+  if (jsonData != null) {
```

Comment:
we should stop with a clear error if no crendetials available.

Reply:
Gradle compiles the full code (while running any task like build) ,which is define inside it. Like we are running the gradle sync here.

![Screenshot from 2023-04-10 11-15-02](https://user-images.githubusercontent.com/34593983/230835001-da2ad30d-695c-4539-8c46-8853a1fd1a35.png)


* So if we want to stop here , then we needs to set the environment variable in other CI tasks.
* For Local machine we can use the pre defined variable ```GITLAB_CI```  to avoid the compilation failure locally https://docs.gitlab.com/ee/ci/variables/predefined_variables.html.

Reply:
compilation and publishing on playstore should be two different targets!

Reply:
@kelson42 , here publishing task depends on compilation. Gradle compiles the full code which is written in it.  I have placed a condition here if it is publishing task, then it gives error if no credentials available in env otherwise it works normally because we are only setting env variable in publishing task on CI.

```
if (gradle.startParameter.taskNames.contains("publishPlayStoreBundle")) {
    val jsonData = System.getenv("PLAYSTORE_JSON")
    if (jsonData != null) {
      jsonFile = file("${buildDir.path}/playstore.json")
      if (!jsonFile.exists()) jsonFile.createNewFile()
      jsonFile.printWriter().use {
        it.print(jsonData)
      }
    } else {
      throw RuntimeException(
        "Please set the environment variable PLAYSTORE_JSON in the CI to publish app " +
          "bundle on PlayStore as it is currently missing"
      )
    }
  }
```

Reply:
@kelson42 , After @gouri-panda review, now we are handing the credentials in CI `Decrypt files` task (to avoid writing extra conditions in android side), if credentials are not available then CI will fail without running any further publication tasks. Like it failed here https://github.com/kiwix/kiwix-android/actions/runs/4628021520/jobs/8186601973.